### PR TITLE
support relative path for rocdl lookup

### PR DESCRIPTION
**Pull request by scchan**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -550,7 +542,7 @@ set(CPACK_DEBIAN_PACKAGE_SECTION "devel")
 set(CPACK_DEBIAN_PACKAGE_CONTROL_EXTRA
   "${PROJECT_BINARY_DIR}/packaging/debian/postinst;${PROJECT_BINARY_DIR}/packaging/debian/prerm")
 
-set(HCC_GENERAL_DEBIAN_DEP "g++-multilib, gcc-multilib, findutils, libelf1, libpci3, file")
+set(HCC_GENERAL_DEBIAN_DEP "coreutils, g++-multilib, gcc-multilib, findutils, libelf1, libpci3, file")
```
Comment (aaronenyeshi) [inline CMakeLists.txt:545]: This will needed to be added to wiki too

File: lib/clamp-device.in [hunk 2]
```diff
@@ -68,12 +68,14 @@ LLD=$BINDIR/ld.lld
 # Determine the ROCm device libs path
 ################
 
-ROCM_DEVICE_LIBS_SEARCH_PATHS="@ROCM_DEVICE_LIB_PATHS@"
+
+HCC_BIN_PATH=$(dirname -- "$(readlink -f -- "$BASH_SOURCE")")
+ROCM_DEVICE_LIBS_SEARCH_PATHS="$HCC_BIN_PATH/../../rocdl/lib;$HCC_BIN_PATH/../rocdl/lib;@ROCM_DEVICE_LIB_PATHS@"
```
Comment (aaronenyeshi) [inline lib/clamp-device.in:73]: Will this check only the binary paths? Should we also check the install paths?

File: lib/clamp-device.in [hunk 3]
```diff
@@ -68,12 +68,14 @@ LLD=$BINDIR/ld.lld
...
 ROCM_LIB=""
 for SEARCH_PATH in $(echo $ROCM_DEVICE_LIBS_SEARCH_PATHS | tr ";" "\n")
 do
   if [ -f "$SEARCH_PATH/ocml.amdgcn.bc" ]; then
-    ROCM_LIB=$SEARCH_PATH
+    ROCM_LIB="$(readlink -f -- "$SEARCH_PATH")"
```
Comment (aaronenyeshi) [inline lib/clamp-device.in:78]: Should we also add the option to set your own environmental variable to the rocdl path you wish?

Comment (scchan) [reply lib/clamp-device.in:78]: nope, it could be done through a cmake option already

File: CMakeLists.txt (hunk 1 above)
Comment (scchan) [reply CMakeLists.txt:545]: done

File: lib/clamp-device.in (hunk 2 above)
Comment (scchan) [reply lib/clamp-device.in:73]: this already covers the common case which is the installation path configured through cmake

**Review by aaronenyeshi — APPROVED** [review]
LGTM

**Pull request merged**